remote/conftest.py

```diff
@@ -0,0 +1,133 @@
+    will be skipped and assumed to be accessible by rho.
+
+    Hosts that are powered on at the beginning of the session are then powered
+    down at the end of the session.
+
+    Example::
+
+        #/home/user/.config/camayoc/config.yaml
+        rho:
+          auths:
+            - username: root
+              name: sonar
+              sshkeyfile: /home/elijah/.ssh/id_sonar_jenkins_rsa
+          hosts:
+            - hostname: string-name-of-vm-on-vcenter
+              ip: XX.XX.XXX.XX
+              provider: vcenter
+              facts:
+                connection.port: 22
+                cpu.count: 1
+          profiles:
+            - name: profile1
+              auths:
+                - auth1
+              hosts:
+                - XX.XX.XXX.XX
+         vcenter:
+          hostname: **Required** -- url to vcenter instance
+          username: **optional** -- will first try environment variable $VCUSER
+          password: **optional** -- will first try environment variable $VCPASS
+
+    *Warning*
+    It is a bad idea to have multiple sessions of pytest running
+    concurrently against the same hosts/config file.
+    If one session ends before another, it will powerdown the
+    machines leaving the other one to fail.
+    """
+    from pyVim.connect import SmartConnect, Disconnect
+    import ssl
+
+    cfg = config.get_config()
+    s = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    s.verify_mode = ssl.CERT_NONE
+    # try and make a secure connection first
+    if 'vcenter' in cfg.keys():
+        if os.getenv('VCUSER', False):
+            vcuser = os.environ['VCUSER']
+        else:
+            vcuser = cfg['vcenter']['username']
+        if os.getenv('VCPASS', False):
+            vcpassword = os.environ['VCPASS']
+        else:
+            vcpassword = cfg['vcenter']['password']
+        try:
+            c = SmartConnect(
+                host=cfg['vcenter']['hostname'],
+                user=vcuser,
+                pwd=vcpassword
+            )
+        except ssl.SSLError:
+            c = SmartConnect(
+                host=cfg['vcenter']['hostname'],
+                user=vcuser,
+                pwd=vcpassword,
+                sslContext=s
+            )
+        # these index choices are particular to our VCenter setup
+        host_folder = c.content.rootFolder.childEntity[DATA_CENTER].hostFolder
+        vm_host = host_folder.childEntity[CLUSTER].host[HOST]
+        # the host has a property "vm" which is a list of VMs
+        vms = vm_host.vm
+        for host in cfg['rho']['hosts']:
+            if ('provider' in host.keys()) and (
+                    host['provider'] == 'vcenter'):
+                for vm in vms:
+                    if vm.name == host['hostname']:
+                        if vm.runtime.powerState == 'poweredOff':
+                            vm.PowerOnVM_Task()
+        # need to wait for machines to boot and start sshd
+        sleep(30)
```

Comment:
We can use sleep now but an improvement would be to pool the machine until an SSH connection can be opened or some state is reached.

Reply:
Yes, I'll file an issue for that. I think that would be a good improvement. A separate loop trying to make an ssh connection after all machines have been sent the powerOn signal would allow them all to boot simultaneously.